### PR TITLE
Renamings in circuit/library/data_encoding

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

Rename the module `qiskit/circuit/library/data_encoding` to `data_preparation` as this is more expressive. Also rename the contents to more descriptive names.

### Details and comments

Suggestions (can be changed):
```
PauliExpansion -> PauliEvolutionFeatureMap
FirstOrderExpansion -> ZExpansionFeatureMap
SecondOrderExpansion -> ZZExpansionFeatureMap
```
Reasons:
* the circuits consist of time evolutions of Pauli strings and are known as "feature maps" in literature
* `pauli_evolution` would be nice because it is short, but these circuits do also the repetitions, entanglements and H-layers on top of the evolutions
* first-order expansion is the special case of the Pauli string `Z`
* second-order expansion is the special case of the Pauli strings `ZZ` and `Z`

### CODE REVIEW

**Comment by jaygambetta** [comment]
what about 
PauliFeatureMap
LocalFeatureMap
TwoLocalFeatureMap

**Comment by Cryoris** [comment]
I'm fine with `PauliFeatureMap` 👍 
`Local` and `TwoLocal` however might be a bit misleading since these classes don't accept arbitrary local and two local gates (for which you would use `TwoLocal`) but really only use `exp(i*t*Z)` and `exp(i*t*ZZ)`. How about `ZFeatureMap` or `PauliZFeatureMap` instead?

**Comment by jaygambetta** [comment]
I'm good with ZFeatureMap and ZZFeatureMap. You are correct for some reason i was thinking i could make it X and XX but since we use the H at the start only Z makes sense.

File: qiskit/circuit/library/__init__.py [hunk 1]
```diff
@@ -161,7 +161,7 @@
    NLocal
    TwoLocal
    RY
-   RYRZ
+   RYRZAnsatz
    SwapRZ
```
Comment (ajavadia) [inline qiskit/circuit/library/__init__.py:165]: only one of these we are going to call Ansatz?

**Pull request merged**